### PR TITLE
minor: Added a javadoc parsing UT

### PR DESCRIPTION
**Pull request by ps-sp**
Taken from [comment](https://github.com/checkstyle/checkstyle/pull/4386#issuecomment-305859660).

### CODE REVIEW

**Comment by ps-sp** [comment]
@romani @Vladlis @rnveach Please have a look. Thansk

**Comment by ps-sp** [comment]
This is the parse tree that's generated:
JAVADOC ->     \r\n* Checks that commentContent starts with '*' javadoc comment identifier.\r\n    *&lt;property name="ignorePattern" value="^ *\* *[^ ]+$"/&gt;\r\n\t* This * is not a leading one\r\n* @return true if commentContent starts with '*' javadoc comment identifier\r\n<EOF> [0:0]
|--TEXT ->      [0:0]
|--NEWLINE -> \r\n [0:4]
|--LEADING_ASTERISK -> * [1:0]
|--TEXT ->  Checks that commentContent starts with '*' javadoc comment identifier. [1:1]
|--NEWLINE -> \r\n [1:72]
|--LEADING_ASTERISK ->     * [2:0]
|--TEXT -> &lt;property name="ignorePattern" value="^ *\* *[^ ]+$"/&gt; [2:5]
|--NEWLINE -> \r\n [2:65]
|--LEADING_ASTERISK -> \t* [3:0]
|--TEXT ->  This * is not a leading one [3:2]
|--NEWLINE -> \r\n [3:30]
|--LEADING_ASTERISK -> * [4:0]
|--WS ->   [4:1]
|--JAVADOC_TAG -> @return true if commentContent starts with '*' javadoc comment identifier\r\n [4:2]
|   |--RETURN_LITERAL -> @return [4:2]
|   |--WS ->   [4:9]
|   `--DESCRIPTION -> true if commentContent starts with '*' javadoc comment identifier\r\n [4:10]
|       |--TEXT -> true if commentContent starts with '*' javadoc comment identifier [4:10]
|       `--NEWLINE -> \r\n [4:75]
`--EOF -> <EOF> [5:0]

**Comment by Vladlis** [comment]
@ps-sp 
Please add an input with `*` in the end of a line

**Comment by rnveach** [comment]
@ps-sp Why did we remove the original test? It was a simple, normal case of asterisks at the beginning.
Your test takes the first one further with asterisks everywhere.
I think the original test should stay.

**Comment by ps-sp** [comment]
@rnveach yeah, but this test has everything that test did. Just the text is different. I felt that there should be only 1 test per element. Right now the UT design doesn't allow it to be done efficiently, but wouldn't it be better if everything relating to one element gets added to the same UT.

**Comment by rnveach** [comment]
@ps-sp Having too much in one UT can also be a bad thing.

> Just the text is different. 

https://github.com/checkstyle/checkstyle/issues/4408 - Just the text is different.
https://github.com/checkstyle/checkstyle/issues/3810 - 1 extra node throws the whole tree off

I don't think multiple tests are a bad thing.

**Comment by romani** [comment]
@ps-sp , to resolve CIs, please rebase.

**Comment by ps-sp** [comment]
@Vladlis 
> Please add an input with * in the end of a line

DONE

@rnveach 
>Why did we remove the original test?

Undone

File: src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/ParseTreeBuilder.java [hunk 1]
```diff
@@ -5794,4 +5794,495 @@ public static ParseTree treeLinkToPackage() {
         javadocContext0.addChild(eof0);
         return javadocContext0;
     }
+
+    public static ParseTree treeLeadingAsterisks2() {
```
Comment (Vladlis) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/ParseTreeBuilder.java]: Please rename in something more meaningful like `treeLeadingAsterisksExtended` (the same for the input file)

File: src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParseTreeTest.java [hunk 2]
```diff
@@ -294,6 +294,15 @@ public void testLinkToPackage() throws IOException {
         compareTrees(expectedTree, generatedTree);
     }
 
+    @Test
+    public void testleadingAsterisks2()
+            throws IOException {
```
Comment (Vladlis) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParseTreeTest.java]: Please move `throws` to the previous line.

File: src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/ParseTreeBuilder.java (hunk 1 above)
Comment (ps-sp) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/ParseTreeBuilder.java]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParseTreeTest.java (hunk 2 above)
Comment (ps-sp) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParseTreeTest.java]: Done

**Comment by Vladlis** [comment]
@ps-sp 
Please also change commit message as now you do not modify a UT, but add a new one

File: src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParseTreeTest.java [hunk 3]
```diff
@@ -294,6 +294,14 @@ public void testLinkToPackage() throws IOException {
         compareTrees(expectedTree, generatedTree);
     }
 
+    @Test
+    public void testleadingAsterisksExtended() throws IOException {
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParseTreeTest.java]: `testleading` => `testLeading`

Comment (ps-sp) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParseTreeTest.java]: Done.

**Comment by Vladlis** [comment]
@romani 
Please do final review

**Pull request merged**